lib/apiservers/portlayer/restapi/handlers/storage_handlers.go

```diff
@@ -648,3 +788,25 @@ func createVsphereVolumeStore(op trace.Operation, dsurl *url.URL, name string, h
 	}
 	return vs, nil
 }
+
+func decodeFilterSpec(filterSpec *string) *archive.FilterSpec {
```

Comment:
The input(to my knowledge) will actually be a map[string]string. so we will need to attempt the decode and the run it through the create function in the archive package(in upstream/master).